tests/integration/util.go

```diff
@@ -1,131 +1,78 @@
+	w := zip.NewWriter(buf)
 
-	var stdout bytes.Buffer
+	// Add some files to the archive.
+	for _, file := range files {
+		f, err := w.Create(file.Name)
+		require.Nil(t, err)
+		_, err = f.Write([]byte(file.Body))
+		require.Nil(t, err)
+	}
 
+	// Make sure to check the error on Close.
+	err := w.Close()
+	require.Nil(t, err)
 
-	cli := &cliTest{
-		MockPrompter: prompter,
-		stdout:       &stdout,
-	}
+	zipBytes, err := ioutil.ReadAll(buf)
+	require.Nil(t, err)
 
-	// Wrap the `PreRun` method, to inject the mock prompter,
-	// and out logger stdout buffer
-	// (global services aren't initialized until `PreRun`
-	preRun := cmd.RootCmd.PersistentPreRunE
-	cmd.RootCmd.PersistentPreRunE = func(c *cobra.Command, a []string) error {
-		// Run the wrapped method
-		err := preRun(c, a)
-		if err != nil {
-			return err
-		}
+	return zipBytes
+}
 
-		// Inject the prompter
-		cmd.Util.Prompter = prompter
 
-		// Tell the logger to log to our stdout buffer
-		logger := cmd.Log.(*observation.LogObservation).LevelLogger.(*logrus.Logger)
-		logger.SetOutput(&stdout)
+// mockEnvVar Sets an env var
+// returns a function to revert the env var back to its previous value
+func mockEnvVar(key, val string) func() {
+	prevVal, ok := os.LookupEnv(key)
+	_ = os.Setenv(key, val)
 
-		// Allow client to inject their own mocks
-		if cli.injector != nil {
-			cli.injector(&injectorInput{cmd.Config, cmd.Service, cmd.Util, cmd.Observation})
+	return func() {
+		if ok {
+			_ = os.Setenv(key, prevVal)
+		} else {
+			_ = os.Unsetenv(key)
 		}
-
-		return nil
 	}
-
-	return cli
 }
 
-func (test *cliTest) Output() string {
-	return test.stdout.String()
-}
 
-type injectorInput struct {
-	config *configs.Root
-	service *service.ServiceContainer
-	util *utl.UtilContainer
-	observation *observation.ObservationContainer
+func stringp(str string) *string {
+	return &str
 }
-type injector func(input *injectorInput)
 
-// Inject allows for injecting dependencies before
-// commands are run. The `inject` function provides
-// pointers to global services used by CLI commands
-//
-// eg
-// cli.Inject(func(input *injectorInput) {
-//		input.util.Weber = &mocks.Weber{}
-// })
-func (test *cliTest) Inject(f injector) {
-	test.injector = f
+func boolp(b bool) *bool {
+	return &b
 }
 
-func writeTempConfig(t *testing.T, conf *configs.Root) string {
-	// Create a tmp file
-	tmpfile, err := ioutil.TempFile("", "dce.*.yml")
-	require.Nil(t, err)
-
-	// Set default config
-	if conf == nil {
-		conf = &configs.Root{
-			API: configs.API{
-				Host:     ptr.String("dce.example.com"),
-				BasePath: ptr.String("/api"),
-			},
-		}
-	}
-
-	// Write config as YAML to tmp file
-	fsUtil := util.FileSystemUtil{
-		Config:     conf,
-		ConfigFile: tmpfile.Name(),
-	}
-	err = fsUtil.WriteConfig()
+// copyStructVals copies all values from the source stuct
+// into the target struct
+// Structs must be JSON encode-able
+func copyStructVals(t *testing.T, source interface{}, target interface{}) {
```

Comment:
We rely pretty heavily on globals in our cobra code, which results in a lot of "leakage" between tests.
This is my attempt to resolve this, by "resetting" global vars to their empty state between tests.